src/ThriftSQL/ResultRow.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace ThriftSQL;
+
+class ResultRow implements \ArrayAccess
+{
+  private $_row;
+  private $_schema;
+
+  public function __construct($schema, $row)
+  {
+    $this->_row = $row;
+    $this->_schema = $schema;
+  }
+
+  public function schemaArray()
+  {
+    $r = [];
+    foreach ($this->_schema as $k=>$i) {
+      $r[$k] = $this->_row[$i];
+    }
+    return $r;
+  }
+
+  private function offsetToIndex($offset)
+  {
+    if(is_int($offset)) {
+      return 0 <= $offset && $offset < count($this->_row);
+    } else if (is_string($offset)) {
+      return ($this->_schema[strtolower($offset)] ?? null);
+    }
+    return null;
+  }
```

Comment:
I don't quite understand the point of this function, its possible return values are `null`, `true`, `false`, or some numeric index that corresponds to the key given. This feels overly complex are you trying to allow access both with key and an index value?

Reply:
Sorry, my bad, i meant `return 0 <= $offset && $offset < count($this->_row) ? $offset : null`